src/init.cpp

```diff
@@ -1158,6 +1159,21 @@ bool AppInitParameterInteraction()
             }
         }
     }
+
```

Comment:
This be done directly in CRegtestParams's constructor, that way, the function UpdateSegwitHeight would not be needed.

Reply:
Seems like a good idea, but segwit height is also updated in one of the unit tests:

https://github.com/bitcoin/bitcoin/pull/12360/files#diff-d5ba361c5f8be78eb4cc0c787c1fc78eR118

I think we need the `UpdateSegwitHeight()` function call there.

Reply:
I guess we still need UpdateSegwitHeight() then, or perhaps we can pass the other args (instead of g_args, or rather g_args plus "-segwitheight=432") to SelectParams in https://github.com/bitcoin/bitcoin/pull/12360/files#diff-d5ba361c5f8be78eb4cc0c787c1fc78eR54 ?
I guess not all BasicTestingSetup want 432 there...

Reply:
I think I'll leave it as is for now. It may be possible to tidy this up in a future commit.